lib/elastomer/client/delete_by_query.rb

```diff
@@ -0,0 +1,82 @@
+module Elastomer
+  class Client
+
+    class DeleteByQuery
+
+      def initialize( client, query, params = nil )
+        @client    = client
+        @query     = query
+        @params    = params
+        @responses = []
+      end
+
+      attr_reader :client, :query, :params, :responses
+
+      def is_ok( status )
+        status.between?(200, 299)
+      end
+
+      def categorize( items )
+        {
+          "found" => items.count { |i| i["found"] },
+          "deleted" => items.count { |i| is_ok(i["status"]) },
+          "missing" => items.count { |i| !i["found"] },
+          "failed" => items.count { |i| i["found"] && !is_ok(i["status"]) },
+        }
+      end
+
+      def accumulate( response )
+        @responses << response unless response == nil
```

Comment:
I will commit the incremental statistics in just a minute, stay tuned.

Good point on the `=` vs `==` mistake - I'll be sure to use the interrogatory form.